File: drivers-simulation/src/main/scala/business/driver/Driver.scala
package io.andrelucas
package business.driver

import java.util.UUID

case class Driver(id: UUID, name: String, modal: String, deliveryRegion: String)

//Companion Pattern
object Driver {
  def create(name: String, modal: String, deliveryRegion: String): Driver = this(UUID.randomUUID(), name, modal, deliveryRegion)
}

File: drivers/src/main/kotlin/io/andrelucas/business/driver/Driver.kt
package io.andrelucas.business.driver

import io.andrelucas.business.region.DeliveryRegion
import io.andrelucas.business.vehicle.VehicleType
import java.util.UUID

data class Driver(val id: UUID = UUID.randomUUID(), val name: String, val vehicleType: VehicleType, val deliveryRegion: DeliveryRegion, val status: DriverStatus = DriverStatus.AVAILABLE)